support/metrics/metricsTracker.go

```diff
@@ -0,0 +1,165 @@
+package metrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/stellar/kelp/support/logger"
+
+	"github.com/google/uuid"
+)
+
+var (
+	amplitudeAPIURL string = "https://api2.amplitude.com/2/httpapi"
+	amplitudeAPIKey string = os.Getenv("AMPLITUDE_API_KEY")
+)
+
+// Tracker wraps the properties for Amplitude events,
+// and can be used to directly send events to the
+// Amplitude HTTP API.
+type Tracker struct {
+	client *http.Client
+	props  props
+	start  time.Time
+}
+
+type event struct {
+	UserID    string      `json:"user_id"`
+	DeviceID  string      `json:"device_id"`
+	EventType string      `json:"event_type"`
+	Props     interface{} `json:"event_properties"`
+}
+
+// props holds the properties that we need for all Amplitude events.
+// This lives on the `Tracker` struct, although
+// TODO: Add geodata.
+// TODO: Add cloud server information.
+type props struct {
+	BuildVersion string    `json:"build_version"`
+	Os           string    `json:"os"`
+	Gui          bool      `json:"gui"`
```

Comment:
Not sure I follow. The spec says that the `gui` data represents "triggered from GUI or not (see --ui flag)". This doesn't reference the GUI version - just how the user is using Kelp.

Should I use the `guiVersion`, rather than `options.gui`?

Reply:
I think I added another comment at the point where this is used, to indicate how to inject in the guiVersion.
Basically you pass in `guiVersion` if `*options.gui` is true, otherwise you pass in `""`.

Yeah, you're right, the spec says it should be a boolean. It would be more useful to have the gui version as well -- we can have both if you think that would be useful, I figured we could achieve the same result by having the `guiVersion` as an empty string for whenever it wasn't triggered by the gui. But it doesn't cost us anything to also include the boolean flag so I'm fine with adding that if you think that will be easier for the query in amplitude.